libs/template/config.go

```diff
@@ -121,6 +121,11 @@ func (c *config) promptForValues() error {
 
 		}
 
+		// Validate the property matches any specified regex pattern.
+		if err := jsonschema.ValidatePatternMatch(name, userInput, property); err != nil {
+			return err
```

Comment:
Can't we let the user input again instead of bailing out?

Reply:
We should be able to do this by calling promptForValues() recursively. I'd prefer to do this as a followup if this is ready to merge as is. WDYT? Created an internal ticket to track.

Reply:
Please prioritize that ticket for the next couple of weeks. The UX of bailing out instead of prompting again is frustrating, especially if the user has already entered more than a couple prompts.

Reply:
Ack, pulled the followup to current sprint.